proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1790 @@
+
+* Otherwise, let `color` be the value of `$color`.
+
+* Let `legacy` be `true` if `$color` is a legacy color and `space` is a
+  legacy color space, and `false` otherwise.
+
+* Let `alpha` be `color`'s alpha property.
+
+* If the keyword argument `$alpha` is specified in `$args`:
+
+    * Set `alpha` to the result of [percent-converting] `$alpha`, and clamping
+      it between 0 and 1 (inclusive).
+
+* Let `channel-args` be the remaining keyword arguments in `$args`, not
+  including `$space` or `$alpha` arguments.
+
+* Let `channels` be a list of the `color`'s channels.
+
+* For each keyword `key` and value `new` in `channel-args`:
+
+  * If `key` is a string in the format `channel<integer>`, set `key` to the
+    value of `<integer>`.
+
+    > This allows e.g. `color.change($color, $channel1: 0.25)` for changing
+    > color channels in unknown color spaces.
+
+  * If `key` is not the name or index of a channel in `channels`, throw an
+    error.
+
+  * Set the corresponding `channel` in `channels` to `new`.
+
+* If `space` is a [known color space], set `channels` to the result of
+  [normalizing] `channels` in `space`.
+
+* Return a color in color space `space`, with `channels` channels, and an alpha
+  of `alpha`.
+
+### `color.adjust()`
+
+```
+adjust($color, $args...)
+```
+
+This function is also available as a global function named `adjust-color()`.
+
+* If `$color` is not a color, throw an error.
+
+* If any item in `$args` is not a keyword argument, throw an error.
+
+* Let `space` be `$color`'s color space.
+
+* If the keyword argument `$space` is specified in `$args`:
+
+    * If `$space` is not a [known color space], and `$space != space`, throw an
+      error.
+
+    * Set `space` to the value of `$space`.
+
+    * Let `color` be the result of [converting] `$color` to `space`.
+
+* Otherwise, let `color` be the value of `$color`.
+
+* Let `legacy` be `true` if `$color` is a legacy color and `space` is a
+  legacy color space, and `false` otherwise.
+
+* Let `alpha` be `color`'s alpha property.
+
+* If the keyword argument `$alpha` is specified in `$args`:
+
+    * If `$alpha` is not a unitless number between `-1` and `1` (inclusive),
+      throw an error.
+
+    * Set `alpha` to the value of `$alpha + alpha` clamped between 0 and 1.
+
+* Let `channel-args` be the remaining keyword arguments in `$args`, not
+  including `$space` or `$alpha` arguments.
+
+* Let `channels` be a list of the `color`'s channels.
+
+* For each keyword `key` and value `adjust` in `channel-args`:
+
+  * If `key` is a string in the format `channel<integer>`, set `key` to the
+    value of `<integer>`.
+
+    > This allows e.g. `color.change($color, $channel1: 0.25)` for changing
+    > color channels in unknown color spaces.
+
+  * If `key` is not the name or index of a channel in `channels`, throw an
+    error.
+
+  * Set the corresponding `channel` in `channels` to `channel + adjust`,
+    treating any `none` keyword as a value of `0`.
```

Comment:
Should we allow percent-conversion here? There's a useful and unambiguous meaning for `color.adjust(red, $red: -20%)`.